ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/EstablishedPeers.hs

```diff
@@ -877,19 +878,18 @@ jobDemoteEstablishedPeer
                           peeraddr peerconn)
 jobDemoteEstablishedPeer PeerSelectionActions {
                            peerStateActions =
-                             PeerStateActions {closePeerConnection}
+                             PeerStateActions {closePeerConnection,
+                                               monitorPeerConnection}
                          }
                          peeraddr peerconn =
     Job job handler () "demoteEstablishedPeer"
   where
     handler :: SomeException
             -> m (Completion m extraState extraDebugState extraFlags extraPeers
                              peeraddr peerconn)
-    handler e = return $
-      -- It's quite bad if closing fails. The peer is cooling so
-      -- we can't remove it from the set of established peers.
-      --
-      Completion $ \st@PeerSelectionState {
+    handler e = do
+      atomically $ monitorPeerConnection peerconn >>= check . (== PeerCold) . fst
```

Comment:
Could you add a haddock explaining why we have to block on the peer to become `cold`.